### PR TITLE
Expose AWS role and user mappings

### PR DESCRIPTION
**Pull request by pgavlin**
Allow additional role and user mappings to be added to the `aws-auth`
config map.

### CODE REVIEW

**Comment by pgavlin** [comment]
cc @eshamay

File: nodejs/eks/cluster.ts [hunk 1]
```diff
@@ -260,15 +311,38 @@ export class Cluster extends pulumi.ComponentResource {
...
         // Enable access to the EKS cluster for worker nodes.
+        const instanceRoleMapping: RoleMapping = {
+            roleArn: instanceRoleARN,
+            username: "system:node:{{EC2PrivateDNSName}}",
+            groups: ["system:bootstrappers", "system:nodes"],
+        };
+        const roleMappings = pulumi.all([pulumi.output(args.roleMappings || []), instanceRoleMapping])
+            .apply(([mappings, instanceMapping]) => {
+                return jsyaml.safeDump([...mappings, instanceMapping].map(m => ({
+                   rolearn: m.roleArn,
```
Comment (hausdorff) [inline nodejs/eks/cluster.ts]: Assuming the `rolearn` casing is correct, is there a reason to not just make `RoleMapping` use the same casing, and doing something like `jsyaml.safeDump([...mappings, instanceMapping])`?

File: nodejs/eks/cluster.ts [hunk 2]
```diff
@@ -17,13 +17,54 @@ import * as k8s from "@pulumi/kubernetes";
...
+
+    /**
+     * The user name within Kubernetes to map to the IAM role. By default, the user name is the ARN of the IAM role.
+     */
+    username: pulumi.Input<string>;
+
+    /**
+     * A list of groups within Kubernetes to which the role is mapped.
+     */
+    groups: pulumi.Input<pulumi.Input<string>[]>;
```
Comment (hausdorff) [inline nodejs/eks/cluster.ts]: Actually, one thing, this type looks like it's on purpose, but it's not clear to me why not just make it `pulumi.Input<string[]>`

Comment (lukehoban) [reply nodejs/eks/cluster.ts]: It's a little clearer if you write `X[]` as `Array<X>` - `pulumi.Input<Array<pulumi.Input<string>>`.  That is, the array itself can be an Output, or the elements of the array can be Outputs.

Comment (hausdorff) [reply nodejs/eks/cluster.ts]: @lukehoban mmm, yes, that makes sense and seems obvious in retrospect that you'd want to do this.

File: nodejs/eks/cluster.ts (hunk 1 above)
Comment (pgavlin) [reply nodejs/eks/cluster.ts]: The `rolearn` casing is correct, yes. I opted to make the user-facing API as idiomatic as possible.

**Pull request merged**